### DISCUSSION TITLE
Elastalert running in docker container does not expose port for Prometheus

### DISCUSSION THREAD

**Discussion by klimarbeit** [jertel/elastalert2#943] [Q&A] [answered]
Hello everyone

When I'm trying to run elastalert in a docker container and expose port for Prometheus, the port does not become open

I read the article about 
"ElastAlert 2 does not have a network API. There is no listening port. You can monitor its activity by viewing the console output or Docker logs", but I think that's not about a special port for Prometheus

example command:
```docker run --net=es_default -d --name elastalert --restart=always -v $(pwd)/elastalert.yaml:/opt/elastalert/config.yaml -v $(pwd)/rules:/opt/elastalert/rules ghcr.io/jertel/elastalert2/elastalert2 --verbose --prometheus_port 9110```

```ss -tulpn``` does not show 9110 listening

```docker ps``` does not show anything in PORTS

I tried to exec in container to watch for ```ss``` or ```netstat```, but command not found and I can't install them because I'm not the root. I know I can reassemble the container and run as a root, but maybe there is another reason why --prometheus_port does not work?

Can anybody help me please

**Comment by jertel** [comment, marked as answer]
The documentation you quoted is attempting to convey the fact that the core ElastAlert 2 system does not listen on a port for API commands. The optional prometheus component does require a listen port, and that documentation can be improved to mention that. In order to expose the port to the host you will need to map the port on the `docker run` command line:

```
docker run --net=es_default -d --name elastalert --restart=always -v $(pwd)/elastalert.yaml:/opt/elastalert/config.yaml -v $(pwd)/rules:/opt/elastalert/rules -p 9110:9110 ghcr.io/jertel/elastalert2/elastalert2 --verbose --prometheus_port 9110
```

As far as exec'ing as root, you do not need to rebuild the container image. You can run:

```
docker exec -it -u root <my-container-id/name> sh
```

**Reply by klimarbeit** [reply]
Thank you very much!
